tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs

```diff
@@ -744,5 +746,87 @@ public void TestJoin()
             Assert.Equal(join["Int_right"][2], right["Int"][2]);
             VerifyJoin(join, left, right, JoinAlgorithm.Inner);
         }
+
+        [Fact]
+        public void TestArrowIntegration()
+        {
+            Field fInt = new Field.Builder().Name("Int").DataType(Int32Type.Default).Build();
+            Field fFloat = new Field.Builder().Name("Float").DataType(FloatType.Default).Build();
+            Schema schema = new Schema.Builder().Field(fInt).Field(fFloat).Build();
+
+            var intArrayBuilder = new Int32Array.Builder().Append(5).Append(10).Append(5).Append(10);
+            Int32Array intArray = intArrayBuilder.Build();
+            var floatArrayBuilder = new FloatArray.Builder().Append(5.0f).Append(10.0f).Append(5.0f).Append(10.0f);
+            FloatArray floatArray = floatArrayBuilder.Build();
+            List<IArrowArray> arrowArrays = new List<IArrowArray>() { intArray, floatArray };
+            RecordBatch recordBatch = new RecordBatch(schema, arrowArrays, 4);
+
+            DataFrame df = new DataFrame(recordBatch);
+            DataFrame df2 = new DataFrame(recordBatch);
+
+            IEnumerable<RecordBatch> recordBatches = df2.AsArrowRecordBatch();
+
+            foreach (RecordBatch batch in recordBatches)
+            {
+                Assert.Equal(batch.ColumnCount, recordBatch.ColumnCount);
```

Comment:
Maybe we could lift the comparison code from the Arrow library?

[https://github.com/apache/arrow/blob/master/csharp/test/Apache.Arrow.Tests/ArrowReaderVerifier.cs#L46](https://github.com/apache/arrow/blob/master/csharp/test/Apache.Arrow.Tests/ArrowReaderVerifier.cs#L46)

Or possibly another way of verifying would be to write the RecordBatch instances to a stream, and then compare that the streams are identical.  Although, debugging issues in that way would be hard.